modules/spark/src/main/scala/org/apache/spark/sql/ignite/IgniteSparkSession.scala

```diff
@@ -63,16 +70,20 @@ class IgniteSparkSession private(ic: IgniteContext, proxy: SparkSession) extends
 
     /** @inheritdoc */
     @transient override lazy val sharedState: SharedState =
-        new IgniteSharedState(ic, sparkContext)
+        existingSharedState.getOrElse(new IgniteSharedState(ic, sparkContext))
 
     /** @inheritdoc */
     @transient override lazy val sessionState: SessionState = {
-        val sessionState = new SessionStateBuilder(self, None).build()
+        parentSessionState
```

Comment:
Currently, we use 4 spaces indentation.
Please, make indentation according to code style